client/my-sites/checkout/checkout/web-payment-box.jsx

```diff
@@ -28,523 +26,383 @@ import {
+		setCanMakePayment( false );
+		const request = stripe.paymentRequest( paymentRequestOptions );
+		request.canMakePayment().then( result => {
+			isSubscribed && setCanMakePayment( !! result );
+		} );
+		request.on( 'paymentmethod', callback );
+		setPaymentRequest( request );
+		return () => ( isSubscribed = false );
+	}, [ stripe, paymentRequestOptions, callback ] );
+
+	return { paymentRequest, canMakePayment };
+}
 
-		if ( ! paymentMethod ) {
-			return null;
-		}
+function LoadingPaymentRequestButton( { paymentType, translate } ) {
+	if ( paymentType === 'apple-pay' ) {
+		return <button className="web-payment-box__apple-pay-button" disabled />;
+	}
+	return (
+		<button
+			className="web-payment-box__web-pay-button button checkout__pay-button-button button is-primary button-pay pay-button__button"
+			disabled
+		>
+			{ translate( 'Select a payment card' ) }
+		</button>
+	);
+}
 
-		const hasBusinessPlanInCart = some( cart.products, ( { product_slug } ) =>
-			overSome( isWpComBusinessPlan, isWpComEcommercePlan )( product_slug )
-		);
-		const showPaymentChatButton = presaleChatAvailable && hasBusinessPlanInCart;
-
-		const buttonState = this.getButtonState();
-		const buttonDisabled = buttonState.disabled;
-		let button;
-		let paymentType;
-
-		switch ( paymentMethod ) {
-			case WEB_PAYMENT_APPLE_PAY_METHOD:
-				if ( buttonState.default ) {
-					button = (
-						<button
-							type="submit"
-							onClick={ event =>
-								! countryCode || ! postalCode
-									? this.displaySubmissionError(
-											translate( 'Please specify a country and postal code.' ),
-											event
-									  )
-									: this.submit( paymentMethod, event )
-							}
-							disabled={ buttonDisabled }
-							className="web-payment-box__apple-pay-button"
-						/>
-					);
-				} else {
-					button = (
-						<Button
-							type="button"
-							className="button is-primary button-pay pay-button__button"
-							busy={ buttonState.disabled }
-							disabled={ buttonDisabled }
-						>
-							{ buttonState.text }
-						</Button>
-					);
-				}
-				paymentType = 'apple-pay';
-				break;
-
-			case WEB_PAYMENT_BASIC_CARD_METHOD:
-				button = (
-					<Button
-						type="submit"
-						className="button is-primary button-pay pay-button__button"
-						onClick={ event => this.submit( paymentMethod, event ) }
-						busy={ buttonState.disabled }
-						disabled={ buttonDisabled }
-					>
-						{ buttonState.text }
-					</Button>
-				);
-				paymentType = 'web-payment';
-				break;
-
-			default:
-				debug( `Unknown payment method ${ paymentMethod }.` );
-		}
+function completePaymentMethodTransaction( {
+	countryCode,
+	postalCode,
+	onSubmit,
+	complete,
+	paymentMethod,
+	payerName,
+} ) {
+	const { card, id } = paymentMethod;
+
+	const newCardDetails = {
+		payment_key: id,
+		name: payerName,
+		country: countryCode,
+		'postal-code': postalCode,
+		card_brand: card.brand,
+		card_display_name: `${ card.brand } ${ card.last4 }`,
```

Comment:
We probably don't need this -- I think the current server-side code that parses it only runs for Paygate anyway.  It's there now just because for raw Apple Pay purchases, the card `displayName` they provide is the only way to get the actual last 4 digits of the user's card.  But it looks like Stripe already takes care of returning that data in a cleaner format in their API.

Reply:
Ah, good find. I had to trace a lot of code to notice that, but I think you're right.

Reply:
Done.